### ISSUE TITLE
Consul request failed with status [400]: Request decode failed: json: unknown field "ServiceWeights"

### ISSUE THREAD

**Issue by goudai** [rickfast/consul-client#436]
# java code
```java
        Consul client = Consul.builder().withUrl("http://127.0.0.1:8500").build();
        AgentClient agentClient = client.agentClient();

        String serviceId = "1";
        Registration service = ImmutableRegistration.builder()
                .id(serviceId)
                .name("myService")
                .port(8080)
                .check(Registration.RegCheck.ttl(3L)) // registers with a TTL of 3 seconds
                .tags(Collections.singletonList("tag1"))
                .serviceWeights(ImmutableServiceWeights.builder().passing(1).warning(5).build())
                .meta(Collections.singletonMap("version", "1.0"))
                .build();

        agentClient.register(service);
        agentClient.pass(serviceId);
```
## docker run 
```bash
docker run     -d     -p 8500:8500     -p 8600:8600/udp     --name=badger     consul agent -server -ui -node=server-1 -bootstrap-expect=1 -client=0.0.0.0
```
exception
```txt
Exception in thread "main" com.orbitz.consul.ConsulException: Consul request failed with status [400]: Request decode failed: json: unknown field "ServiceWeights"
	at com.orbitz.consul.util.Http.ensureResponseSuccessful(Http.java:61)
	at com.orbitz.consul.util.Http.handle(Http.java:39)
	at com.orbitz.consul.AgentClient.register(AgentClient.java:214)
	at com.orbitz.consul.AgentClient.register(AgentClient.java:218)
```

**Comment by kkonstantin42** [comment]
We are experiencing the same issue. That's happening because the field should be called 'Weights', not 'ServiceWeights' (at least in the latest consul version): https://www.consul.io/api-docs/agent/service#register-service
The error is also reproducible by running integration tests, e.g.:
`shouldSetWarning(com.orbitz.consul.AgentITest)  Time elapsed: 0.003 sec  <<< ERROR!
com.orbitz.consul.ConsulException: Consul request failed with status [400]: Request decode failed: json: unknown field "ServiceWeights"
        at com.orbitz.consul.AgentITest.shouldSetWarning(AgentITest.java:310)

`
I am happy to submit a pull request with a fix.

**Comment by dmelemed** [comment]
I am also experiencing this issue. Thanks for working on a fix.

**Comment by rickfast** [comment]
This is fixed in 1.5.1
